Build error

Is this error something I'm doing wrong?

```
   Compiling prometheus_wireguard_exporter v3.6.4 (/usr/local/share/prometheus_wireguard_exporter)
error[E0277]: the trait bound `ServerOptions: From<(IpAddr, u16)>` is not satisfied
   --> src/main.rs:223:16
    |
223 |     let addr = (ip, bind).into();
    |                ^^^^^^^^^^ ---- required by a bound introduced by this call
    |                |
    |                the trait `From<(IpAddr, u16)>` is not implemented for `ServerOptions`
    |
    = note: required for `(IpAddr, u16)` to implement `Into<ServerOptions>`

error[E0277]: `ServerOptions` doesn't implement `std::fmt::Display`
   --> src/main.rs:225:53
    |
225 |     info!("starting exporter on http://{}/metrics", addr);
    |                                                     ^^^^ `ServerOptions` cannot be formatted with the default formatter
    |
    = help: the trait `std::fmt::Display` is not implemented for `ServerOptions`
    = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
    = note: this error originates in the macro `format_args` which comes from the expansion of the macro `info` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `prometheus_wireguard_exporter` due to 2 previous errors
error: failed to compile `prometheus_wireguard_exporter v3.6.4 (/usr/local/share/prometheus_wireguard_exporter)`, intermediate artifacts can be found at `/usr/local/share/prometheus_wireguard_exporter/target`
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name                     = "prometheus_wireguard_exporter"
-version                  = "3.6.4"
+version                  = "3.6.5"
 authors                  = ["Francesco Cogno <francesco.cogno@outlook.com>"]
 description              = "Prometheus WireGuard Exporter"
 edition                  = "2018"

--- a/README.md
+++ b/README.md
@@ -4,11 +4,11 @@
 
 [![Crate](https://img.shields.io/crates/v/prometheus_wireguard_exporter.svg)](https://crates.io/crates/prometheus_wireguard_exporter) [![cratedown](https://img.shields.io/crates/d/prometheus_wireguard_exporter.svg)](https://crates.io/crates/prometheus_wireguard_exporter) [![cratelastdown](https://img.shields.io/crates/dv/prometheus_wireguard_exporter.svg)](https://crates.io/crates/prometheus_wireguard_exporter)
 
-[![release](https://img.shields.io/github/release/MindFlavor/prometheus_wireguard_exporter.svg)](https://github.com/MindFlavor/prometheus_wireguard_exporter/tree/3.6.4)
-[![tag](https://img.shields.io/github/tag/mindflavor/prometheus_wireguard_exporter.svg)](https://github.com/MindFlavor/prometheus_wireguard_exporter/tree/3.6.4)
+[![release](https://img.shields.io/github/release/MindFlavor/prometheus_wireguard_exporter.svg)](https://github.com/MindFlavor/prometheus_wireguard_exporter/tree/3.6.5)
+[![tag](https://img.shields.io/github/tag/mindflavor/prometheus_wireguard_exporter.svg)](https://github.com/MindFlavor/prometheus_wireguard_exporter/tree/3.6.5)
 
 [![Rust build](https://github.com/mindflavor/prometheus_wireguard_exporter/workflows/Rust/badge.svg)](https://github.com/mindflavor/prometheus_wireguard_exporter/actions?query=workflow%3ARust)
-[![commitssince](https://img.shields.io/github/commits-since/mindflavor/prometheus_wireguard_exporter/3.6.4)](https://img.shields.io/github/commits-since/mindflavor/prometheus_wireguard_exporter/3.6.4)
+[![commitssince](https://img.shields.io/github/commits-since/mindflavor/prometheus_wireguard_exporter/3.6.5)](https://img.shields.io/github/commits-since/mindflavor/prometheus_wireguard_exporter/3.6.5)
 
 [![Docker build](https://github.com/MindFlavor/prometheus_wireguard_exporter/actions/workflows/docker.yml/badge.svg)](https://github.com/qdm12/godevcontainer/actions/workflows/docker.yml)
 

--- a/src/main.rs
+++ b/src/main.rs
@@ -2,6 +2,7 @@ use anyhow::Context;
 use clap::{crate_authors, crate_name, crate_version, value_parser, Arg};
 use hyper::{Body, Request};
 use log::{debug, info, trace};
+use prometheus_exporter_base::prelude::{Authorization, ServerOptions};
 use std::env;
 mod options;
 use options::Options;
@@ -220,11 +221,16 @@ async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
 
     let bind: u16 = *matches.get_one("port").unwrap();
     let ip: IpAddr = *matches.get_one("addr").unwrap();
-    let addr = (ip, bind).into();
+    let addr: std::net::SocketAddr = (ip, bind).into();
 
     info!("starting exporter on http://{}/metrics", addr);
 
-    render_prometheus(addr, options, |request, options| {
+    let server_options = ServerOptions {
+        addr,
+        authorization: Authorization::None,
+    };
+
+    render_prometheus(server_options, options, |request, options| {
         Box::pin(perform_request(request, options))
     })
     .await;

--- a/src/wireguard.rs
+++ b/src/wireguard.rs
@@ -36,6 +36,7 @@ impl From<&str> for SecureString {
     }
 }
 
+#[allow(dead_code)]
 #[derive(Default, Debug, Clone)]
 pub(crate) struct LocalEndpoint {
     pub public_key: String,
@@ -53,6 +54,7 @@ pub(crate) struct RemoteEndpoint {
     pub latest_handshake: u64,
     pub sent_bytes: u128,
     pub received_bytes: u128,
+    #[allow(dead_code)]
     pub persistent_keepalive: bool,
 }
 

--- a/src/wireguard_config.rs
+++ b/src/wireguard_config.rs
@@ -8,6 +8,7 @@ use std::convert::TryInto;
 #[derive(Debug, Default, Clone)]
 pub(crate) struct PeerEntry<'a> {
     pub public_key: &'a str,
+    #[allow(dead_code)]
     pub allowed_ips: &'a str,
     pub friendly_description: Option<FriendlyDescription<'a>>,
 }

